src/main/java/com/conveyal/datatools/manager/jobs/MergeLineContext.java

```diff
@@ -0,0 +1,899 @@
+                keyValue = csvReader.get(keyFieldIndex);
+                // When all stops missing stop_code for the first feed, there's nothing to do (i.e.,
+                // no failure condition has been triggered yet). Just indicate this in the flag and
+                // proceed with the merge.
+                if (handlingFutureFeed) stopCodeMissingFromFutureFeed = true;
+                // However... if the second feed was missing stop_codes and the first feed was not,
+                // fail the merge job.
+                else if (!stopCodeMissingFromFutureFeed) {
+                    job.failMergeJob(
+                        stopCodeFailureMessage(stopsMissingStopCodeCount, stopsCount, specialStopsCount)
+                    );
+                }
+            } else if (stopsMissingStopCodeCount > 0) {
+                // If some, but not all, stops are missing stop_code, the merge feeds job must fail.
+                job.failMergeJob(
+                    stopCodeFailureMessage(stopsMissingStopCodeCount, stopsCount, specialStopsCount)
+                );
+            }
+        }
+    }
+
+    private boolean shouldCheckStopCodes() {
+        return job.mergeType.equals(SERVICE_PERIOD) && table.name.equals(STOPS);
+    }
+
+    /** Determine if stop is "special" via its locationType. I.e., a station, entrance, (location_type > 0). */
+    private boolean isSpecialStop(String locationType) {
+        return !"".equals(locationType) && !"0".equals(locationType);
+    }
+
+    public boolean updateAgencyIdIfNeeded() {
+        if (newAgencyId != null && field.name.equals(AGENCY_ID) && job.mergeType.equals(REGIONAL)) {
+            if (val.equals("") && table.name.equals("agency") && lineNumber > 0) {
+                // If there is no agency_id value for a second (or greater) agency
+                // record, return null which will trigger a failed merge feed job.
+                job.failMergeJob(String.format(
+                    "Feed %s has multiple agency records but no agency_id values.",
+                    feed.version.id
+                ));
+                return false;
+            }
+            LOG.info("Updating {}#agency_id to (auto-generated) {} for ID {}", table.name, newAgencyId, keyValue);
+            val = newAgencyId;
+        }
+        return true;
+    }
+
+    public void startNewField(int specFieldIndex) throws IOException {
+        this.field = sharedSpecFields.get(specFieldIndex);
+        // Get index of field from GTFS spec as it appears in feed
+        index = fieldsFoundList.indexOf(field);
+        val = csvReader.get(index);
+        // Default value to write is unchanged from value found in csv (i.e. val). Note: if looking to
+        // modify the value that is written in the merged file, you must update valueToWrite (e.g.,
+        // updating this feed's end_date or accounting for cases where IDs conflict).
+        valueToWrite = val;
+    }
```

Comment:
It might be a good idea to create a subclass of FieldContext to store these values that are specifically related to fields.

Reply:
See [Extract Subclass](https://refactoring.guru/extract-subclass).

Reply:
`FieldContext` extracted in https://github.com/ibi-group/datatools-server/pull/376/commits/1b57a05108dd842f91d8992aa30af969d1f8a719.